engine/include/cubos/engine/gizmos/gizmos.hpp

```diff
@@ -0,0 +1,114 @@
+/// @file
+/// @brief Resource @ref cubos::engine::Gizmos.
+/// @ingroup gizmos-plugin
+
+#pragma once
+#include <vector>
+
+#include <glm/glm.hpp>
+
+#include <cubos/core/gl/render_device.hpp>
+
+#include <cubos/engine/cubos.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Resource that holds the information needed to draw a gizmo
+    class GizmosRenderer final
+    {
+    public:
+        cubos::core::gl::ShaderPipeline pipeline;    ///< Shader pipeline to be used when drawing gizmos
+        cubos::core::gl::RenderDevice* renderDevice; ///< Active render device
+
+        /// @brief Set of buffers and structs that hold te information needed to draw a specific type of mesh
+        struct Primitive
+        {
+            cubos::core::gl::VertexBuffer vb;
+            cubos::core::gl::VertexArray va;
+            cubos::core::gl::IndexBuffer ib;
+            cubos::core::gl::VertexArrayDesc vaDesc;
+        };
+
+        Primitive linePrimitive; ///< GL line information
+        Primitive boxPrimitive;  ///< GL box information
+
+        /// @brief Sets up the GizmosRenderer to be used
+        /// @param renderDevice the current RenderDevice being used
+        void init(cubos::core::gl::RenderDevice* currentRenderDevice);
+
+    private:
+        void initLinePrimitive();
+        void initBoxPrimitive();
+    };
```

Comment:
Move this resource to a header in the src/ dir, as its never used outside.

Reply:
Y'know, i considered doing that, but dismissed it as it was a resource. I forgot you could just include the local header in the plugin.cpp too :)